geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/query/QueryProcess.scala

```diff
@@ -98,6 +102,10 @@ class QueryVisitor(features: SimpleFeatureCollection,
     source.getFeatures(query)
   }
 
+  override def getExpressions: java.util.List[Expression] = {
+    val (_, finalSFT) = QueryPlanner.buildTransformSFT(origSft, properties)
+    FilterHelper.propertyNames(filter, finalSFT).map(ff.property).toList
```

Comment:
this should include both the props from the filter and the props from the sft.
you don't have to pass the finalSFT into the propertyNames function, i'm not sure that would do the right thing (it expects the original sft)